From 5e85cfb1f6232ed010db90ce42da1bdf02bafcfa Mon Sep 17 00:00:00 2001
From: Konrad Leszczynski <konrad.leszczynski@intel.com>
Date: Thu, 27 Aug 2015 11:34:44 +0100
Subject: [PATCH] Disable the HLE (Hardware LPM enable) for the host
 controller.

By setting the hw_lpm_support to 0 in xhci structure,
all the subsequent flows for setting the HLE will not succeed,
therefore the Hardware LPM Enable will be disabled.

Change-Id: Iff0b44d49959a700713c0e7196d4f57a2146ab12
Signed-off-by: Konrad Leszczynski <konrad.leszczynski@intel.com>
---
 drivers/usb/host/xhci-mem.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/host/xhci-mem.c b/drivers/usb/host/xhci-mem.c
index f8893b3..e17ff65 100644
--- a/drivers/usb/host/xhci-mem.c
+++ b/drivers/usb/host/xhci-mem.c
@@ -2016,8 +2016,8 @@ static void xhci_add_in_port(struct xhci_hcd *xhci, unsigned int num_ports,
 		xhci->sw_lpm_support = 1;
 		if (temp & XHCI_HLC) {
 			xhci_dbg_trace(xhci, trace_xhci_dbg_init,
-					"xHCI 1.0: support USB2 hardware lpm");
-			xhci->hw_lpm_support = 1;
+					"xHCI 1.0: don't support USB2 hardware lpm");
+			xhci->hw_lpm_support = 0;
 		}
 	}
 
-- 
1.9.1

